src/stores/ontRuns.js

```diff
@@ -97,13 +106,45 @@ export const useOntRunsStore = defineStore('ontRuns', {
         const ontRootStore = useOntRootStore()
         const existingInstruments = ontRootStore.instruments
 
-        //TODO: This need to be refactored to use the Pinia once ont/pools is migrated
-        const existingPools = store.getters['traction/ont/pools/pools']
+        const { pools } = groupIncludedByResource(included)
+        this.pools = formatById(this.pools, pools, true)
 
-        this.currentRun = buildFormatedOntRun(existingInstruments, existingPools, data, included)
+        this.currentRun = buildFormatedOntRun(existingInstruments, this.pools, data, included)
         return { success, errors }
       }
     },
+
+    /**
+     * Fetches a pool by its barcode and adds the pool to the store if it exists.
+     *
+     * @param {string} barcode - The barcode of the pool to fetch.
+     * @returns {Promise<{success: boolean}>} - An object indicating the success of the operation.
+     */
+    async fetchPool(barcode) {
+      // Here we want to make sure the barcode exists
+      // If it doesn't, set success to null for component validation
+      if (!barcode || barcode.trim() === '') {
+        return {
+          success: true,
+        }
+      }
+      const rootStore = useRootStore()
+      const request = rootStore.api.traction.ont.pools
+      const promise = request.get({ filter: { barcode } })
+      const response = await handleResponse(promise)
+      let { success, body: { data } = {} } = response
+
+      // If response is successful and the
+      if (success && data.length > 0) {
```

Comment:
I see this all over the place it makes me feel that we should add another flag to the response handler to check that there is data.

Don't think we need to do it here but maybe put a note on it.